tests/rule_based_profiler/data_assistant/test_onboarding_data_assistant.py

```diff
@@ -1042,3 +1049,119 @@ def test_onboarding_data_assistant_plot_metrics_stdout(
+                    "class_name": "RuntimeDataConnector",
+                    "batch_identifiers": ["default_identifier_name"],
+                },
+                "default_inferred_data_connector_name": {
+                    "class_name": "InferredAssetSqlDataConnector",
+                    "include_schema_name": True,
+                },
+            },
+        }
+
+        context.add_datasource(**datasource_config)
+        batch_request = RuntimeBatchRequest(
+            datasource_name="my_trino_datasource",
+            data_connector_name="default_runtime_data_connector_name",
+            data_asset_name="default_name",  # this can be anything that identifies this data
+            runtime_parameters={
+                "query": "SELECT pickup_datetime, dropoff_datetime, store_and_fwd_flag from taxi_data LIMIT 10"
+            },
+            batch_identifiers={"default_identifier_name": "default_identifier"},
+        )
+    else:
+        datasource = context.sources.add_sql(
+            name="my_trino_datasource", connection_string=CONNECTION_STRING
+        )
+        asset = datasource.add_query_asset(
+            name="default_name",
+            query="SELECT pickup_datetime, dropoff_datetime, store_and_fwd_flag from taxi_data LIMIT 10",
+        )
+        batch_request = asset.build_batch_request()
+
+    context.add_or_update_expectation_suite(expectation_suite_name="test_suite")
+    validator = context.get_validator(
+        batch_request=batch_request, expectation_suite_name="test_suite"
+    )
+
+    desired_metric = MetricConfiguration(
+        metric_name="table.column_types",
+        metric_domain_kwargs={},
+        metric_value_kwargs=None,
+    )
+    validator.execution_engine.resolve_metrics([desired_metric])
+
+    if not use_fds:
+        batch_ids = get_batch_ids(data_context=context, batch_request=batch_request)
+        # Ensure that the data types are read correctly
+        semantic_type_filter = SimpleSemanticTypeFilter(
+            validator=validator,
+            batch_ids=batch_ids,
+            column_names=["pickup_datetime", "dropoff_datetime", "store_and_fwd_flag"],
+        )
+        assert (
+            semantic_type_filter.table_column_name_to_inferred_semantic_domain_type_map
+            == {
+                "pickup_datetime": SemanticDomainTypes.DATETIME,
+                "dropoff_datetime": SemanticDomainTypes.TEXT,
+                "store_and_fwd_flag": SemanticDomainTypes.TEXT,
+            }
+        )
+
+    # Attempt to run the onboarding assistant
+    result = context.assistants.onboarding.run(
+        batch_request=batch_request,
+        estimation="flag_outliers",
+        include_column_names=[
+            "pickup_datetime",
+            "dropoff_datetime",
+            "store_and_fwd_flag",
+        ],
+    )
+
+    assert result
+    assert len(result.expectation_configurations) > 0
+    assert len(result.rule_exception_tracebacks) == 0
```

Comment:
When using FDS, this fails due to inability to get columns from BatchData. I'm not sure why this is happening. Could whomever picks this up look into this? Perhaps I have something set wrong for FDS?

Reply:
Looking now

Reply:
Update:  Do you know if the issue is because we are using a `QueryAsset` or a query to define our batch? I'm a bit puzzled why we are getting an error for both the `FDS` and `non-FDS` cases.